### PR TITLE
TSK-1085: Hateoas self-links which contain variables are not working …

### PR DESCRIPTION
**Pull request by nkolytschew**
…for Paged representationmodels

* remove Aspect
* remove PageLinks
* remove AbstractRessourcesAssembler
* update TaskanaPagingAssembler
* update *Assembler implementing TaskanaPagingAssembler for self-links

* update unit-test and test "href-self" part of json
* fix checkstyl-finding

### CODE REVIEW

**Comment by nkolytschew** [comment]
Sonar https://sonarcloud.io/dashboard?branch=feature%2Fnko-1085_refactor-aspect&id=nkolytschew_taskana
Duplication detected on classes I didn't touch 🙈

Travis likes it tho: https://travis-ci.org/github/nkolytschew/taskana/builds/715167273

File: rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/assembler/TaskCommentRepresentationModelAssembler.java [hunk 1]
```diff
@@ -70,9 +68,9 @@ public TaskanaPagedModelKeys getProperty() {
   }
 
   @Override
-  @PageLinks(Mapping.URL_TASK_COMMENTS)
   public TaskanaPagedModel<TaskCommentRepresentationModel> toPageModel(
       Iterable<TaskComment> taskComments, PageMetadata pageMetadata) {
-    return TaskanaPagingAssembler.super.toPageModel(taskComments, pageMetadata);
+    return addLinksToPageResource(
+        TaskanaPagingAssembler.super.toPageModel(taskComments, pageMetadata), pageMetadata);
```
Comment (holgerhagen) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/assembler/TaskCommentRepresentationModelAssembler.java:74]: Why do we add the metadata twice and do not simply get it from the model?

File: history/taskana-simplehistory-rest-spring/src/main/java/pro/taskana/simplehistory/rest/assembler/TaskHistoryEventListResourceAssembler.java [hunk 2]
```diff
@@ -4,53 +4,57 @@
...
 
-import pro.taskana.resource.rest.AbstractRessourcesAssembler;
 import pro.taskana.simplehistory.impl.HistoryEventImpl;
 import pro.taskana.simplehistory.rest.TaskHistoryEventController;
 import pro.taskana.simplehistory.rest.models.TaskHistoryEventListResource;
 import pro.taskana.simplehistory.rest.models.TaskHistoryEventRepresentationModel;
 
 /** Mapper to convert from a list of HistoryEventImpl to a TaskHistoryEventResource. */
-public class TaskHistoryEventListResourceAssembler extends AbstractRessourcesAssembler {
+public class TaskHistoryEventListResourceAssembler {
```
Comment (holgerhagen) [inline history/taskana-simplehistory-rest-spring/src/main/java/pro/taskana/simplehistory/rest/assembler/TaskHistoryEventListResourceAssembler.java:23]: I understand that we do not change the history events to the representation model structure in this PR? Is there a ticket for it to do it later on?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/assembler/TaskanaPagingAssembler.java [hunk 3]
```diff
@@ -25,4 +33,46 @@
   default TaskanaPagedModel<D> toPageModel(Iterable<T> entities) {
     return toPageModel(entities, null);
   }
+
+  default TaskanaPagedModel<D> addLinksToPageResource(
```
Comment (holgerhagen) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/assembler/TaskanaPagingAssembler.java:37]: would addLinksToPage'd'Resource be the better name?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/assembler/TaskCommentRepresentationModelAssembler.java (hunk 1 above)
Comment (nkolytschew) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/assembler/TaskCommentRepresentationModelAssembler.java:74]: You're right! Thank you!

**Pull request merged**